Ferlab-Ste-Justine/Post-processing-Pipeline
# fix: CLIN-3706 support more input gvcf file extensions

This pull request introduces a new step at the beginning of the workflow (BCFTOOLS VIEW) to standardize the input VCF files.

The current implementation will fail if the file extension is .gvcf in a solo family and if the exclude mnps step is skipped. This issue is caused by an imperfection in the GENOTYPEGVCFS module (from nf-core) that makes an incorrect assumption about file extensions. Specifically, it incorrectly assumes that the VCF file is a gendb url when the extension is not .vcf.gz.

To fix this problem and prevent other data compatibility issues with file extensions and index files, we introduce a new step at the beginning of the workflow (BCFTOOLS_VIEW) to standardize the input VCF files. The vcf files will be saved with a .g.vcf.gz extension, which is standard in nf-core processes. If the index file is not present, it will be generated.

**Tests**

1) Run the pipeline locally:
nextflow run main.nf -profile test,docker
-The exclude mnps step should be executed
-Check that the output files of the process BCFTOOLS_VIEW
-Check that the bodies of the final output files are identical to those from obtain via the main branch

2) Reproduce the file extension bug locally using the main branch.  Then switch to the PR branch and check that the pipeline runs successfully. 
- The exclude mnp step should be  skipped
-Compare the results with initial dataset (with .g.vcf.gz extensions) and check that the body of the splitMultiAllelics output is identical.

3) Remove the .tbi files and check that the pipeline still runs successfully without exclude mnps

**Tests in juno**

Run the pipeline normally
-Check that exclude mnp step is executed
- the pipeline should be executed successfully

Run the pipeline with .gvcf.gz files instead .g.vcf.gz files and no .tbi files, exclude_mnps set to false
- the exclude mnps process should be skipped
- the pipeline should be executed successfully



## PR checklist

- [ ] Reference Data Documentation in `docs/reference_data.md` is updated.
